src/clib/step_rate_newton_raphson.hpp

```diff
@@ -402,277 +479,30 @@ inline void step_rate_newton_raphson(
-           my_rates->SN0_fFe3O4, my_rates->SN0_fAC, my_rates->SN0_fSiO2D, my_rates->SN0_fMgO,
-           my_rates->SN0_fFeS, my_rates->SN0_fAl2O3,
-           my_rates->SN0_freforg, my_rates->SN0_fvolorg, my_rates->SN0_fH2Oice,
-           my_rates->SN0_r0SiM, my_rates->SN0_r0FeM, my_rates->SN0_r0Mg2SiO4, my_rates->SN0_r0MgSiO3,
-           my_rates->SN0_r0Fe3O4, my_rates->SN0_r0AC, my_rates->SN0_r0SiO2D, my_rates->SN0_r0MgO,
-           my_rates->SN0_r0FeS, my_rates->SN0_r0Al2O3,
-           my_rates->SN0_r0reforg, my_rates->SN0_r0volorg, my_rates->SN0_r0H2Oice,
-           my_rates->gr_N, &my_rates->gr_Size, &my_rates->gr_dT, my_rates->gr_Td,
-           my_rates->SN0_kpSiM, my_rates->SN0_kpFeM, my_rates->SN0_kpMg2SiO4, my_rates->SN0_kpMgSiO3,
-           my_rates->SN0_kpFe3O4, my_rates->SN0_kpAC, my_rates->SN0_kpSiO2D, my_rates->SN0_kpMgO,
-           my_rates->SN0_kpFeS, my_rates->SN0_kpAl2O3,
-           my_rates->SN0_kpreforg, my_rates->SN0_kpvolorg, my_rates->SN0_kpH2Oice,
-           my_rates->h2dustS, my_rates->h2dustC, my_rates->grain_growth_rate,
-           &grain_temperatures.data[OnlyGrainSpLUT::SiM_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::FeM_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::Mg2SiO4_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::MgSiO3_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::Fe3O4_dust][i],
-           &grain_temperatures.data[OnlyGrainSpLUT::AC_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::SiO2_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::MgO_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::FeS_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::Al2O3_dust][i],
-           &grain_temperatures.data[OnlyGrainSpLUT::ref_org_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::vol_org_dust][i], &grain_temperatures.data[OnlyGrainSpLUT::H2O_ice_dust][i],
-           my_rates->gas_grain2, &my_rates->gamma_isrf2,
-           &imp_eng[i],
-           &my_chemistry->radiative_transfer_HDI_dissociation, &kdissHDI(i,j,k), &my_chemistry->radiative_transfer_metal_ionization, &kphCI(i,j,k), &kphOI(i,j,k),
-           &my_chemistry->radiative_transfer_metal_dissociation, &kdissCO(i,j,k), &kdissOH(i,j,k), &kdissH2O(i,j,k),
-           &my_chemistry->radiative_transfer_use_H2_shielding, &my_chemistry->H2_custom_shielding, &f_shield_custom(i,j,k)
-          );
+          // calc the time derivatives
+          t_deriv::derivatives(dt_FIXME, dsp.data(), dspdot.data(), pack);
 
           for (jsp = 1; jsp<=(nsp); jsp++) {
-            dspj = eps * dsp[idsp[jsp-1]-1];
+            dspj = eps * dsp[idsp[jsp-1]];
             for (isp = 1; isp<=(nsp); isp++) {
               if(isp == jsp)  {
-                dsp1[idsp[isp-1]-1] = dsp[idsp[isp-1]-1] + dspj;
+                dsp1[idsp[isp-1]] = dsp[idsp[isp-1]] + dspj;
               } else {
-                dsp1[idsp[isp-1]-1] = dsp[idsp[isp-1]-1];
+                dsp1[idsp[isp-1]] = dsp[idsp[isp-1]];
```

Comment:
Just checking, what's happening here with the removal of the -1s? The for loops are the same, so just want to understand.